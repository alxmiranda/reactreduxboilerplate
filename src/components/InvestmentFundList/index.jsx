import React from 'react';
import { PropTypes } from 'prop-types';

const InvestmentFundList = ({ loading, investmentFunds }) => (
  <div>
    <h1>Fundos de investimento</h1>
    {loading && <h2>Carregando...</h2>}
    {!loading && !investmentFunds.length && <p>Nenhum fundo encontrado.</p>}
    {!loading && !!investmentFunds.length &&
      <table>
        <thead>
          <tr>
            <th>Code</th>
            <th>Name</th>
            <th>Manager</th>
          </tr>
        </thead>
        <tbody>
          {investmentFunds.map(fund => (
            <tr key={fund.id}>
              <td>{fund.code}</td>
              <td>{fund.name}</td>
              <td>{fund.fundManager}</td>
            </tr>))}
        </tbody>
      </table>}
  </div>
);

InvestmentFundList.defaultProps = {
  loading: false,
  investmentFunds: [],
};

InvestmentFundList.propTypes = {
  loading: PropTypes.bool,
  investmentFunds: PropTypes.arrayOf(PropTypes.object),
};

export default InvestmentFundList;
