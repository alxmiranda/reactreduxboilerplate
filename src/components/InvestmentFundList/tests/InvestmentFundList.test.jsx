import React from 'react';
import { render } from 'enzyme';
import expect from 'expect';
import InvestmentFundList from '../';

const investmentFunds = [
  {
    code: '6029',
    cnpj: '12865128000105',
    name: 'CAPITÂNIA EQUITIES INSTITUCIONAL FUNDO DE INVESTIMENTO EM COTAS DE FUNDOS DE INVESTIMENTO EM AÇÕES',
    id: '0b8dd692-94d2-4f34-bea4-0073eca1718a',
    fundManagerId: 'c2f66b68-5f78-4a06-b3c3-f9007e0529c7',
    fundManager: 'Capitania Investimentos',
    properties: {
      investmentFundId: '0b8dd692-94d2-4f34-bea4-0073eca1718a', id: 'e2d5916e-846a-4d98-4f2e-08d50048c8fe', active: true, privateCredit: false, accountOrder: false, xpPlatform: true, qualifiedInvestor: false,
    },
    productId: 1,
    productName: 'Fundos de investimento',
    brands: [{ id: 3, name: 'Xp Investimentos' }],
  }, {
    code: '6935',
    cnpj: '16498956000103',
    name: 'Modal Equity Value FIA',
    id: '74df40e8-34c7-478f-bf21-0094e2059de9',
    fundManagerId: '97f449f2-4926-4284-b2ae-f6e5c1862527',
    fundManager: 'Modal Asset',
    properties: {
      investmentFundId: '74df40e8-34c7-478f-bf21-0094e2059de9', id: '068c4ff4-8297-4076-4f8a-08d50048c8fe', active: true, privateCredit: false, accountOrder: false, xpPlatform: true, qualifiedInvestor: false,
    },
    productId: 1,
    productName: 'Fundos de investimento',
    brands: [{ id: 3, name: 'Xp Investimentos' }],
  }, {
    code: '770000',
    cnpj: '17685620000104',
    name: 'Brasil Plural Prev FIC FIM',
    id: '37292c9f-9787-4b4f-8629-00f117709f73',
    fundManagerId: '86bffbaa-2f6b-4e65-b714-eb3b115f0820',
    fundManager: 'Brasil Plural',
    properties: {
      investmentFundId: '37292c9f-9787-4b4f-8629-00f117709f73', id: '3fbee6e7-fcba-423d-505e-08d50048c8fe', active: true, privateCredit: false, accountOrder: true, xpPlatform: true, qualifiedInvestor: false,
    },
    productId: 3,
    productName: 'Fundo de previdência',
    brands: [{ id: 3, name: 'Xp Investimentos' }],
  },
];

describe('InvestmentFundList', () => {
  it('Renders as the snapshot if no prop is provided', () => {
    const rendered = render(<InvestmentFundList />);
    expect(rendered).toMatchSnapshot();
  });

  it('Renders as the snapshot if it is loading', () => {
    const rendered = render(<InvestmentFundList loading />);
    expect(rendered).toMatchSnapshot();
  });

  it('Renders as the snapshot if it has investmentFunds', () => {
    const rendered = render(<InvestmentFundList investmentFunds={investmentFunds} />);
    expect(rendered).toMatchSnapshot();
  });

  it('Renders as the snapshot if it has investmentFunds and also is loading', () => {
    const rendered = render(<InvestmentFundList loading investmentFunds={investmentFunds} />);
    expect(rendered).toMatchSnapshot();
  });
});
