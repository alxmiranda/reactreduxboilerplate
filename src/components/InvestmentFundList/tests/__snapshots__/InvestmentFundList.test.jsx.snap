// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvestmentFundList Renders as the snapshot if it has investmentFunds 1`] = `
<div>
  <h1>
    Fundos de investimento
  </h1>
  <table>
    <thead>
      <tr>
        <th>
          Code
        </th>
        <th>
          Name
        </th>
        <th>
          Manager
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          6029
        </td>
        <td>
          CAPITÂNIA EQUITIES INSTITUCIONAL FUNDO DE INVESTIMENTO EM COTAS DE FUNDOS DE INVESTIMENTO EM AÇÕES
        </td>
        <td>
          Capitania Investimentos
        </td>
      </tr>
      <tr>
        <td>
          6935
        </td>
        <td>
          Modal Equity Value FIA
        </td>
        <td>
          Modal Asset
        </td>
      </tr>
      <tr>
        <td>
          770000
        </td>
        <td>
          Brasil Plural Prev FIC FIM
        </td>
        <td>
          Brasil Plural
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`InvestmentFundList Renders as the snapshot if it has investmentFunds and also is loading 1`] = `
<div>
  <h1>
    Fundos de investimento
  </h1>
  <h2>
    Carregando...
  </h2>
</div>
`;

exports[`InvestmentFundList Renders as the snapshot if it is loading 1`] = `
<div>
  <h1>
    Fundos de investimento
  </h1>
  <h2>
    Carregando...
  </h2>
</div>
`;

exports[`InvestmentFundList Renders as the snapshot if no prop is provided 1`] = `
<div>
  <h1>
    Fundos de investimento
  </h1>
  <p>
    Nenhum fundo encontrado.
  </p>
</div>
`;
